language/go/generate.go

```diff
@@ -37,10 +37,20 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) language.GenerateRes
 	c := args.Config
 	gc := getGoConfig(c)
 	pcMode := getProtoMode(c)
+
+	// This is a collection of go_proto_library rules that were generated by
+	// another gazelle rule.
+	//
+	//   key: proto_library rule name (colon-prefixed)
+	//   value: go_proto_library rule
+	goProtoRules := make(map[string]*rule.Rule)
 	var protoRuleNames []string
 	protoPackages := make(map[string]proto.Package)
 	protoFileInfo := make(map[string]proto.FileInfo)
 	for _, r := range args.OtherGen {
+		if r.Kind() == "go_proto_library" {
+			goProtoRules[r.AttrString("proto")] = r
```

Comment:
Let's be very careful here.

* `proto` may not be set (`AttrString` may return `""`).
* `protos` may be set instead (it should be a list of strings).
* Either may be set to a list of labels that aren't necessarily in the same directory.
* More than one `go_proto_library` may reference the same `proto_library`.